tests/dragonfly/http_conf_test.py

```diff
@@ -1,77 +1,132 @@
+        "admin_nopass": True,
+    }
+)
+async def test_no_password_on_admin(df_server: DflyInstance):
+    async with get_http_session("default", "XXX") as session:
+        resp = await session.get(f"http://localhost:{df_server.admin_port}/")
+        assert resp.status == 200
+    async with get_http_session("random") as session:
+        resp = await session.get(f"http://localhost:{df_server.admin_port}/")
+        assert resp.status == 200
+    async with get_http_session() as session:
+        resp = await session.get(f"http://localhost:{df_server.admin_port}/")
+        assert resp.status == 200
+
+
+@dfly_args({"proactor_threads": "1", "requirepass": "XXX", "admin_port": 1113})
+async def test_password_on_admin(df_server: DflyInstance):
+    async with get_http_session("default", "badpass") as session:
+        resp = await session.get(f"http://localhost:{df_server.admin_port}/")
+        assert resp.status == 401
+    async with get_http_session() as session:
+        resp = await session.get(f"http://localhost:{df_server.admin_port}/")
+        assert resp.status == 401
+    async with get_http_session("default", "XXX") as session:
+        resp = await session.get(f"http://localhost:{df_server.admin_port}/")
+        assert resp.status == 200
+
+
+@dfly_args({"proactor_threads": "1", "expose_http_api": "true"})
+async def test_no_password_on_http_api(df_server: DflyInstance):
+    async with get_http_session("default", "XXX") as session:
+        resp = await session.post(f"http://localhost:{df_server.port}/api", json=["ping"])
+        assert resp.status == 200
+    async with get_http_session("random") as session:
+        resp = await session.post(f"http://localhost:{df_server.port}/api", json=["ping"])
+        assert resp.status == 200
+    async with get_http_session() as session:
+        resp = await session.post(f"http://localhost:{df_server.port}/api", json=["ping"])
+        assert resp.status == 200
+
+
+@dfly_args({"proactor_threads": "1", "expose_http_api": "true"})
+async def test_http_api(df_server: DflyInstance):
+    client = df_server.client()
+    async with get_http_session() as session:
+        async with session.get(f"http://localhost:{df_server.port}") as resp:
             assert resp.status == 200
-        async with aiohttp.ClientSession() as session:
-            resp = await session.get(f"http://localhost:{server.admin_port}/")
+
+        body = '["set", "foo", "МайяХилли", "ex", "100"]'
+        async with session.post(f"http://localhost:{df_server.port}/api", data=body) as resp:
             assert resp.status == 200
+            text = await resp.text()
+            assert text.strip() == '{"result":"OK"}'
 
+        body = '["get", "foo"]'
+        async with session.post(f"http://localhost:{df_server.port}/api", data=body) as resp:
+            assert resp.status == 200
+            text = await resp.text()
+            assert text.strip() == '{"result":"МайяХилли"}'
```

Comment:
content-type json is not required?

Reply:
currently no, we just try parsing the data. will add a todo.